slycot/analysis.py

```diff
@@ -783,81 +804,83 @@ def ab09ax(dico,job,n,m,p,A,B,C,nr=None,tol=0.0,ldwork=None):
-            `nr` is set as follows:
-            If on input ``nr is not None``, `nr` is equal to ``MIN(nr,NMIN)``,
-            where `nr` is the desired order on entry and `NMIN` is the order
-            of a minimal realization of the given system; `NMIN` is
-            determined as the number of Hankel singular values greater
-            than ``n*eps*HNORM(A,B,C)``, where `eps` is the machine
-            precision (see LAPACK Library Routine DLAMCH) and
-            ``HNORM(A,B,C)`` is the Hankel norm of the system (computed
-            in ``HSV(1)``);
-            If on input ``nr is None``, `nr` is equal to the number of Hankel
-            singular values greater than ``MAX(tol,n*eps*HNORM(A,B,C))``.
-        Ar : rank-2 array('d') with bounds ``(nr,nr)``
-            This array contains the state dynamics matrix `Ar` of the reduced
-            order system.
-        Br : rank-2 array('d') with bounds ``(nr,m)``
-            Tthis array contains the input/state matrix `Br` of the reduced
-            order system.
-        Cr : rank-2 array('d') with bounds ``(p,nr)``
-            This array contains the state/output matrix `Cr` of the reduced
-            order system.
-        hsv : output double precision array, dimension ``(n)``
-            If ``INFO = 0``, it contains the Hankel singular values of
-            the original system ordered decreasingly. ``HSV(1)`` is the
-            Hankel norm of the system.
-        T : rank-2 array('d') with bounds ``(n,nr)``
-            This array contains the right truncation matrix `T` of the reduced
-            order system.
-        Ti : rank-2 array('d') with bounds ``(nr,n)``
-            This array contains the left truncation matrix `Ti` of the reduced
-            order system.
+    Parameters
+    ----------
+    dico : {'D', 'C'}
+        Indicate whether the system is discrete `D` or continuous `C`
+    job : {'B', 'N'}
+        Balance `B` or not `N`
+    n : int
+        The number of state variables. n >= 0.
+    m : int
+        The number of system inputs. m >= 0.
+    p : int
+        The number of system outputs. p >= 0.
+    A : (n, n) array_like
+        The leading n-by-n part of this array must contain the state
+        dynamics matrix A of the system *in real Schur form.*
+    B : (n, m) array_like
+        The leading n-by-m part of this array must contain the input/state
+        matrix B of the system.
+    C : (p, n) array_like
+        The leading p-by-n part of this array must contain the
+        state/output matrix C of the system.
+    nr : int, optional
+        `nr` is the desired order of the resulting reduced order
+        system.  ``0 <= nr <= n``. Automatically determined by `tol` if
+        ``nr is None`` and returned. See return object `nr`.
+        Default is None.
+    tol : double precision, optional
+        If ``nr is None``, `tol`contains the tolerance for determining the
```

Comment:
```suggestion
        If `nr` is `None`, `tol` contains the tolerance for determining the
```

Reply:
It is true, there is a difference between ``monospace`` (not for variables) and `single back-ticks` (for variable, module, function) in the numpydoc style (https://numpydoc.readthedocs.io/en/latest/format.html#common-rest-concepts).

```markdown
For paragraphs, indentation is significant and indicates indentation in the output. New paragraphs are marked with a blank line.

Use *italics*, **bold** and ``monospace`` if needed in any explanations (but not for variable names and doctest code or multi-line code). Variable, module, function, and class names should be written between single back-ticks (`numpy`).

A more extensive example of reST markup can be found in [this example document](http://docutils.sourceforge.net/docs/user/rst/demo.txt); the [quick reference](http://docutils.sourceforge.net/docs/user/rst/quickref.html) is useful while editing.

Line spacing and indentation are significant and should be carefully followed.
```

I think this kind of numpydoc correction should be done after the merge of the four basic clean-ups (math, transform, analysis, synthesis).  The use of `` `` and ` ` is all over slycot.

I would prefer to fix the pytest file-structure first and the main numpydoc organization. After the merge of all 4 open PR's we can do smaller numpydoc fixes.

Reply:
Okay, but please add the space after `tol` now.